watertap/unit_models/selective_oil_permeation.py

```diff
@@ -0,0 +1,669 @@
+        self.add_port(name="permeate", block=self.properties_permeate)
+
+        # References for control volume
+        # pressure change
+        if (
+            self.config.has_pressure_change is True
+            and self.config.momentum_balance_type != "none"
+        ):
+            self.deltaP = Reference(self.feed_side.deltaP)
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="Average transmembrane pressure",
+        )
+        def eq_pressure_transmemb(b, t):
+            return (
+                b.pressure_transmemb[t]
+                == 0.5
+                * (
+                    b.feed_side.properties_in[t].pressure
+                    + b.feed_side.properties_out[t].pressure
+                )
+                - b.properties_permeate[t].pressure
+            )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.config.property_package.phase_list,
+            self.config.property_package.component_list,
+            doc="Mass transfer term",
+        )
+        def eq_mass_transfer_term(b, t, p, j):
+            if j == "H2O":
+                self.feed_side.mass_transfer_term[t, p, "H2O"].fix(
+                    0
+                )  # no water transfer
+                return Constraint.Skip
+            elif j == "oil":
+                return (
+                    b.mass_transfer_oil[t]
+                    == -b.feed_side.mass_transfer_term[t, p, "oil"]
+                )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="Oil mass transfer",
+        )
+        def eq_oil_transfer(b, t):
+            return (
+                b.flux_vol_oil[t]
+                * b.feed_side.properties_in[0].dens_mass_phase_comp["Liq", "oil"]
+                * b.area
+                == -b.feed_side.mass_transfer_term[t, "Liq", "oil"]
+            )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.config.property_package.component_list,
+            doc="Permeate production",
+        )
+        def eq_permeate_production(b, t, j):
+            return (
+                b.properties_permeate[t].get_material_flow_terms("Liq", j)
+                == -b.feed_side.mass_transfer_term[t, "Liq", j]
+            )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="Isothermal assumption for permeate",
+        )
+        def eq_permeate_isothermal(b, t):
+            return (
+                b.feed_side.properties_in[t].temperature
+                == b.properties_permeate[t].temperature
+            )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="Shell side iquid velocity",
+        )
+        def eq_liquid_velocity(b, t):
+            return (
+                b.feed_side.properties_in[t].flow_vol_phase["Liq"]
+                == b.liquid_velocity[t] * 0.25 * Constants.pi * b.module_diameter**2
```

Comment:
If `liquid_velocity` doesn't get modified to include inlet and outlet conditions, I'd rename to `liquid_velocity_in`